src/commands/distribute/release.ts

```diff
@@ -0,0 +1,224 @@
+import { AppCommand, CommandResult, ErrorCodes, failure, hasArg, help, isCommandFailedResult, longName, required, shortName, success} from "../../util/commandline";
+import { MobileCenterClient, models, clientRequest } from "../../util/apis";
+import { out } from "../../util/interaction";
+import * as _ from "lodash";
+import * as Request from "request";
+import * as Path from "path";
+import * as Pfs from "../../util/misc/promisfied-fs";
+import { DefaultApp } from "../../util/profile";
+
+const debug = require("debug")("mobile-center-cli:commands:distribute:release");
+
+@help("Upload release binary and trigger distribution")
+export default class ReleaseBinaryCommand extends AppCommand {
+  @help("Path to binary file")
+  @shortName("f")
+  @longName("file")
+  @required
+  @hasArg
+  public filePath: string;
+
+  @help("Distribution group name")
+  @shortName("g")
+  @longName("group")
+  @required
+  @hasArg
+  public distributionGroup: string;
+
+  @help("Release notes text")
+  @shortName("r")
+  @longName("release-notes")
+  @hasArg
+  public releaseNotes: string;
+
+  @help("Path to release notes file")
+  @shortName("R")
+  @longName("release-notes-file")
+  @hasArg
+  public releaseNotesFile: string;
+
+  public async run(client: MobileCenterClient): Promise<CommandResult> {
+    try {
+      return await this.doCommand(client);
+    } catch (error) {
+      if (isCommandFailedResult(error)) {
+        return error;
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  private async doCommand(client: MobileCenterClient): Promise<CommandResult> {
+    const app: DefaultApp = this.app;
+    
+    debug("Check that user have selected either --release-notes or --release-notes-file");
+    this.validateParameters();
+
+    debug("Loading prerequisites");
+    const [distributionGroupUsers, releaseBinaryFileBuffer, releaseNotesString] = await out.progress("Loading prerequisites...", this.getPrerequisites(client));
+
+    debug("Creating release upload");
+    const createdReleaseUpload = await this.createReleaseUpload(client, app);
+    const uploadUri = createdReleaseUpload.uploadUrl;
+    const uploadId = createdReleaseUpload.uploadId;
+
+    let releaseUrl: string;
+    try {
+      debug("Uploading release binary");
+      await out.progress("Uploading release binary...", this.uploadFileToUri(uploadUri, releaseBinaryFileBuffer, Path.basename(this.filePath)));
+
+      debug("Finishing release upload");
+      releaseUrl = await this.finishReleaseUpload(client, app, uploadId);
+    } catch (error) {
+      try {
+        out.text("Release upload failed");        
+        await this.abortReleaseUpload(client, app, uploadId);
+        out.text("Release Upload was aborted");
+      } catch (abortError) {
+        out.text("Failed to abort release upload");
```

Comment:
This is a really funny message. I don't think we should notify the user about it. 
I would just say keep the original "Release Upload was aborted".